sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/Uri.java

```diff
@@ -183,7 +184,25 @@ public boolean shouldRefreshHealthStatus() {
     }
 
     public String getHealthStatusDiagnosticString() {
-        return this.healthStatusTuple.get().diagnsoticString;
+        return this.healthStatusTuple.get().getDiagnsoticString();
+    }
+
+    public void setHealthStatusTupleAttempting(boolean attempting) {
```

Comment:
whether the replica is attempted or not is based on the flow. Updating here might cause unnecessary contention. I think the places to decided whether SDK will attempt it or not will be during the flow

Reply:
Changed it to be set in the replicaStatusList in ConsistencyWriter and StoreReader